pallets/gear-builtin/src/benchmarking.rs

```diff
@@ -311,6 +315,24 @@ benchmarks! {
         let standard = <ark_bls12_381::g2::Config as SWCurveConfig>::mul_projective(&base, &bigint);
         assert_eq!(standard, result.0);
     }
+
+    staking_estimate_decode {
+        let l in 1 .. MAX_PAYLOAD_SIZE as u32;
+
+        // Longest payload corresponds to the `RequestV1::Nominate` variant.
```

Comment:
consider limitations on payload length and pre-check of it, since amount of nominees is limited with 16 (I'd even consider using LimitedVec for 16 hashes, but seems too much since this param is dictated by the runtime and governance)
so max payload len for this is 1 + 4 + 18 * 32 bytes atm

Reply:
it will remove need in benching anything there

Reply:
That's correct that a valid nominate request should comply with Runtime settings, however nothing stops a caller contract send an ill-formed request (that will inevitably be rejected in the upstream extrinsic) which we still need to decode. So unless we use a `LimitedVec` in the API (as you suggest, with some reasonable limit that we will never exceed in the runtime) estimating the decoding cost based on the payload size seems appropriate, is it not?